ruma-events-macros/src/event_parse.rs

```diff
@@ -31,6 +32,7 @@ pub enum EventKind {
     Message(Ident),
     State(Ident),
     ToDevice(Ident),
+    ManuallyImpled(Ident),
```

Comment:
This is the escape hatch for `PresenceEvent` and the `RedactionEvent`s. Do you think this is ok, or do you have another idea?

Reply:
I think I would prefer the variant here to be separated from the stored ident (so `to_event_ident` gets one more parameter, maybe it should no longer be a method too), and for places that have to deal with events that fit none of the kinds to take `Option<EventKind>`.

Reply:
I'm not sure I follow this. The `to_event_ident` would be
```rust
enum EventKind {
    Message,
    State,
    ManuallyImpled,
}
fn to_event_ident(kind: &EventKind, var: &EventKIndVariant, ident: &ident) -> Option<Ident> {}
```

Reply:
No, I mean all functions where `ManuallyImpled` would be a possible input should instead take `Option<EventKind>` instead of `EventKind` parameters and `EventKind` should not have a `ManuallyImpled` variant. To allow still having the ident currently contained in `EventKind` in the `None` case, split it from the enum altogether.

Reply:
Does it make sense now?